frame/nomination-pools/src/lib.rs

```diff
@@ -582,19 +582,200 @@ pub struct PoolRoles<AccountId> {
+	pub throttle: Option<CommissionThrottle<T>>,
+}
+
+impl<T: Config> Default for Commission<T> {
+	fn default() -> Self {
+		Self { current: None, max: None, throttle: None }
+	}
+}
+
+impl<T: Config> Commission<T> {
+	/// Get the current commission percentage of this pool.
+	/// Returns zero if commission has not yet been configured.
+	fn commission_or_zero(&self) -> Perbill {
+		self.current.as_ref().map(|(x, _)| *x).unwrap_or(Perbill::zero())
+	}
+
+	/// Returns true if a commission percentage updating to `to` would exhaust the throttle limit.
+	///
+	/// A commission update will be throttled (disallowed) if:
+	/// 1. not enough blocks have passed since the previous commission update took place, and
+	/// 2. the new commission is larger than the maximum allowed increase.
+	///
+	/// Throttling is not applied to commission updates if `current` is still `None`.
+	fn throttling(&self, to: &Perbill) -> bool {
+		let throttle = self.throttle.as_ref().map(|t| t).or(None);
+		if let Some(t) = throttle {
+			// check for `min_delay` throttling
+			if t.previous_set_at.map_or(false, |p| {
+				<frame_system::Pallet<T>>::block_number().saturating_sub(p) <
+					t.change_rate.min_delay
+			}) {
+				return true
+			}
+			// check for `max_increase` throttling
+			(*to).saturating_sub(self.commission_or_zero()) > t.change_rate.max_increase &&
+				self.current.is_some()
+		} else {
+			return false
+		}
+	}
+
+	/// Set the pool's commission.
+	/// Update commission accordingly based on `commission` and `payee`.
+	/// If throttle is present, record the current block as the previously updated commission.
+	/// If the supplied commission is zero, `None` will be inserted and `payee` will be ignored.
+	fn maybe_update_current(
+		&mut self,
+		commission: &Perbill,
+		payee: T::AccountId,
+	) -> DispatchResult {
+		ensure!(!self.throttling(&commission), Error::<T>::CommissionChangeThrottled);
+		ensure!(
+			self.max.map(|m| commission <= &m).unwrap_or(true),
+			Error::<T>::CommissionExceedsMaximum
+		);
+
+		self.current = Some((*commission, payee));
+
+		// update `throttle.previous_set_at` if a throttle has been configured.
+		let _ = self
+			.throttle
+			.as_mut()
+			.map(|t| t.register_change(<frame_system::Pallet<T>>::block_number()));
+		Ok(())
+	}
+
+	/// Set the pool's maximum commission.
+	/// The pool's maximum commission can be set to any value initially, and only smaller values
+	/// thereafter. If larger values are attempted, this function will return a dispatch error.
+	///
+	/// If `current.0` is larger than an updated max commission value, then `current.0` will also be
+	/// updated to the new maximum.
+	fn maybe_update_max(&mut self, new_max: Perbill) -> DispatchResult {
+		if let Some(old) = self.max.as_mut() {
+			if new_max > *old {
+				return Err(Error::<T>::MaxCommissionRestricted.into())
+			}
+			*old = new_max;
+			let _ = self.current.as_mut().map(|(x, _)| *x = (*x).min(new_max));
+		} else {
+			self.max = Some(new_max)
+		};
```

Comment:
```suggestion
		} else {
			self.max = Some(new_max)
		};
		let _ = self.current.as_mut().map(|(x, _)| *x = (*x).min(new_max));
```

If max is not set, my current is 10%, and I set max to 5%, my current is not adjusted. This is a bug, before fixing it please make sure there is a test for it.